### PR TITLE
DropDown and Segmented control

### PR DESCRIPTION
**Pull request by PavelLaptev**
- segmented control — centered icons
- some CSS fixes
- updated `DropDown` component. Now the component uses `Button` component. Did it in order to reduce code duplication.

### CODE REVIEW

File: packages/ui/src/lib/components/DropDown.svelte [hunk 1]
```diff
@@ -51,88 +59,47 @@
...
+		&:hover {
+			z-index: 2;
 		}
 	}
 
-	.label {
-		display: inline-flex;
-		padding: 0 var(--space-2);
-	}
+	.dropdown-wrapper :global(.dropdown__icon-btn) {
```
Comment (mtsgrd) [inline packages/ui/src/lib/components/DropDown.svelte:77]: Could we try and do this differently so we don't have to rely on `:global()`?

File: packages/ui/src/lib/components/DropDown.svelte [hunk 2]
```diff
@@ -51,88 +59,47 @@
...
+		z-index: 1;
```
Comment (mtsgrd) [inline packages/ui/src/lib/components/DropDown.svelte:78]: Why is this necessary?

File: packages/ui/src/styles/text-classes.css [hunk 3]
```diff
@@ -119,6 +119,7 @@
 	font-weight: 700;
 	line-height: 100%;
 }
+sdf
```
Comment (mtsgrd) [inline packages/ui/src/styles/text-classes.css:122]: This looks accidental.

File: packages/ui/src/lib/utils/joinClassNames.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function joinClassNames(classNames: string[]): string {
+	return classNames.filter(Boolean).join(' ');
```
Comment (mtsgrd) [inline packages/ui/src/lib/utils/joinClassNames.ts:2]: What is this?

File: packages/ui/src/lib/icons/Icon.svelte [hunk 5]
```diff
@@ -12,11 +12,8 @@
 </script>
 
 <svg
-	class={className}
+	class={`icon-wrapper ${className}`}
```
Comment (mtsgrd) [inline packages/ui/src/lib/icons/Icon.svelte:15]: Let's discuss this at some point. I think I'm slowly turning against this pattern, there seems to always be a cleaner solution imho.

File: packages/ui/src/lib/components/DropDown.svelte (hunk 1 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/components/DropDown.svelte:77]: Because Svelte isolates class names, you can't pass a class from one component to another, like you do for example in React. But don't worry, this `:global()` thing will only affect this component. Other components with the same names won't be affected. It's isolated to this component.

Here is an example: https://svelte.dev/repl/c16aa7a6ef714067af6f5eee0a0f453c?version=3.29.0

File: packages/ui/src/lib/utils/joinClassNames.ts (hunk 4 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/utils/joinClassNames.ts:2]: I use it here:
image

I tried to add classes like this `class:outlined={kind}` but seems like it doesn't work for components
image

File: packages/ui/src/lib/icons/Icon.svelte (hunk 5 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/icons/Icon.svelte:15]: Yes, let's discuss this tomorrow 🙂

File: packages/ui/src/lib/utils/joinClassNames.ts (hunk 4 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/utils/joinClassNames.ts:2]: We can change the approach. It's just how I did it here.

File: packages/ui/src/lib/components/DropDown.svelte (hunk 1 above)
Comment (mtsgrd) [reply packages/ui/src/lib/components/DropDown.svelte:77]: The point I'm making isn't about working/not working, or affecting other components, but rather about readability and maintainability. It gets difficult to reason about the style of components when classes are passed in dynamically.

File: packages/ui/src/styles/text-classes.css (hunk 3 above)
Comment (PavelLaptev) [reply packages/ui/src/styles/text-classes.css:122]: yes. that's already removed

File: packages/ui/src/lib/components/DropDown.svelte (hunk 2 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/components/DropDown.svelte:78]: Yes, I need to bring the hovered button to the top

image

It's because I'm overlapping one outline button with another

image

File: packages/ui/src/lib/utils/joinClassNames.ts (hunk 4 above)
Comment (mtsgrd) [reply packages/ui/src/lib/utils/joinClassNames.ts:2]: Oh sorry, that was too ambiguous. I was looking at filtering using `Boolean`. At first I didn't understand how it's related, but I see it's for filtering out any empty/undefined items. But this doesn't add value here, and imho more appropriate when you're working on a version library used by other people. Let's talk about this one too over ☕.
